internal/internal_task_handlers.go

```diff
@@ -558,6 +558,7 @@ func (wth *workflowTaskHandlerImpl) createWorkflowContext(task *workflowservice.
 	if taskQueue == nil || taskQueue.Name == "" {
 		return nil, errors.New("nil or empty TaskQueue in WorkflowExecutionStarted event")
 	}
+	task.Messages = append(inferMessages(task.GetHistory().GetEvents()), task.Messages...)
```

Comment:
Why do we need to append here? shouldn't `Messages` be empty?

Reply:
If there was one or more messages on the WFT that initiated this whole thing then that message won't be represented by a WorkflowUpdateAccepted event in the history so we want to propagate it along with the inferred messages

Reply:
For any given instance of Update, yes. But you could have three UpdateAccepted in the history _plus_ an update.Request message (not yet accepted) on the WFT. In this case you would want to synthesize update.Request messages for replay _and also_ preserve the newer as-yet-unaccepted request that does not exist in history.